devtools/create_project/xcode.cpp

```diff
@@ -25,6 +25,7 @@
 
 #include <fstream>
 #include <algorithm>
+#include <sys/stat.h>
```

Comment:
I am by no means an expert on Xcode project files, but I don’t think so? My understanding is Xcode project files are just plist files and have no capability for specifying events like “when this project file is opened, run this command/code/whatever”. The closest thing would be an external build target set up as a prerequisite for the actual build target (I think CMake does this), but I am not sure if is safe to modify the project build file in the middle of a build (I don’t think Xcode will pick up the changes until the next build), so it wouldn’t provide a very good user experience. Detecting here doesn’t seem any different than running autoconf to generate Makefiles with the right paths though?

Some other options I can think of are:

1. Unconditionally specify all the directories in search paths. This will cause warning to be emitted at build time for the ones that don’t exist (which is why I didn’t just do that to begin with). The existing code was also using an explicit path for the static libraries, I’m not sure offhand if a relative filename + libraries search path is good enough? (I would think so but I really was more concerned about just fixing the defines-issue than making this generator excellent.)
2. Add `--prefix` flag to the generator that will override the path search if it is specified.